### ISSUE TITLE
Annotation serialization and parsing

### ISSUE THREAD

**Issue by jmakeig** [jmakeig/markdown-annotations#23]
Annotations should be tracked in the same file as the text of the Markdown and any optional metadata. Need a way to serialize annotations into a Markdown document such that

  * It doesn’t create a diff on the *main content*
  * It’s unambiguous to parse

**Comment by jmakeig** [comment]
Pandoc, for example, supports block of [YAML metadata](https://pandoc.org/MANUAL.html#extension-yaml_metadata_block). Start with this concept, but JSON?

**Comment by jmakeig** [comment]
Also, [Rangy](https://github.com/timdown/rangy/wiki/Highlighter-Module).

**Comment by jmakeig** [comment]
Tried [ProseMirror](http://prosemirror.net) annotations, but this looks like a lot of work and would use their Markdown interface. Although, this path would lead to a place where the Markdown would be editable eventually.

**Comment by jmakeig** [comment]
[Treora/dom-highlight-range](https://github.com/Treora/dom-highlight-range)

**Comment by jmakeig** [comment]
Fixed by #28. 

Ranges are stored in the client state, relative to the Markdown itself, not how it’s rendered in the DOM. This makes the annotations durable across different DOM representations. 

```json
{
  "annotations": [
    {
      "timestamp": "2017-11-04T05:24:12.283Z",
      "comment": "",
      "user": "jmakeig",
      "range": {
        "start": {
          "row": 92,
          "column": 573
        },
        "end": {
          "row": 96,
          "column": 697
        }
      }
    },
    {
      "timestamp": "2017-11-04T05:23:56.577Z",
      "comment": "",
      "user": "jmakeig",
      "range": {
        "start": {
          "row": 116,
          "column": 1046
        },
        "end": {
          "row": 116,
          "column": 1317
        }
      }
    },
    {
      "timestamp": "2017-11-04T05:24:06.268Z",
      "comment": "",
      "user": "jmakeig",
      "range": {
        "start": {
          "row": 116,
          "column": 1479
        },
        "end": {
          "row": 118,
          "column": 519
        }
      }
    },
    {
      "timestamp": "2017-11-04T05:24:01.658Z",
      "comment": "",
      "user": "jmakeig",
      "range": {
        "start": {
          "row": 118,
          "column": 331
        },
        "end": {
          "row": 118,
          "column": 617
        }
      }
    }
  ],
  "href": "https://github.com/…",
  "commit": "SHA…",
  "content": "…"
}
```
